pinot-common/src/main/java/org/apache/pinot/common/request/context/LiteralContext.java

```diff
@@ -0,0 +1,126 @@
+import org.apache.pinot.common.request.Literal;
+import org.apache.pinot.spi.data.FieldSpec;
+
+
+/**
+ * The {@code LiteralContext} class represents a literal in the query.
+ * <p>This includes both value and type information. We translate thrift literal to this representation in server.
+ * Currently, only Boolean literal is correctly encoded in thrift and passed in.
+ * All integers are encoded as LONG in thrift, and the other numerical types are encoded as DOUBLE.
+ * The remaining types are encoded as STRING.
+ */
+public class LiteralContext {
+  // TODO: Support all of the types for sql.
+  private FieldSpec.DataType _type;
+  private Object _value;
+
+  private static FieldSpec.DataType convertThriftTypeToDataType(Literal._Fields fields) {
+    switch (fields) {
+      case LONG_VALUE:
+        return FieldSpec.DataType.LONG;
+      case BOOL_VALUE:
+        return FieldSpec.DataType.BOOLEAN;
+      case DOUBLE_VALUE:
+        return FieldSpec.DataType.DOUBLE;
+      case STRING_VALUE:
+        return FieldSpec.DataType.STRING;
+      default:
+        throw new UnsupportedOperationException("Unsupported literal type:" + fields);
+    }
+  }
+
+  private static Class<?> convertDataTypeToJavaType(FieldSpec.DataType dataType) {
+    switch (dataType) {
+      case INT:
+        return Integer.class;
+      case LONG:
+        return Long.class;
+      case BOOLEAN:
+        return Boolean.class;
+      case FLOAT:
+        return Float.class;
+      case DOUBLE:
+        return Double.class;
+      case STRING:
+        return String.class;
+      default:
+        throw new UnsupportedOperationException("Unsupported dataType:" + dataType);
+    }
+  }
+
+  public LiteralContext(Literal literal) {
+    _type = convertThriftTypeToDataType(literal.getSetField());
+    _value = literal.getFieldValue();
+  }
+
+  public FieldSpec.DataType getType() {
+    return _type;
+  }
+
+  @Nullable
+  public Object getValue() {
+    return _value;
+  }
+
+  public LiteralContext(FieldSpec.DataType type, Object value) {
+    Preconditions.checkArgument(convertDataTypeToJavaType(type) == value.getClass(),
+        "Unmatched data type: " + type + " java type: " + value.getClass());
+    _type = type;
+    _value = value;
+  }
+
+  @Override
+  public int hashCode() {
+    if (_value == null) {
+      return 31 * 31 * _type.hashCode();
+    }
+    return 31 * 31 * _type.hashCode() + 31 * _value.hashCode();
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof LiteralContext)) {
+      return false;
+    }
+    LiteralContext that = (LiteralContext) o;
+    return _type.equals(that._type) && Objects.equals(_value, that._value);
+  }
+
+  @Override
+  public String toString() {
+    if (_value == null) {
+      return '\'' + "" + '\'';
+    }
+    // TODO: print out the type.
+    if(_type == FieldSpec.DataType.STRING || _type == FieldSpec.DataType.BYTES) {
```

Comment:
What is the value for `BYTES` type? If it is `byte[]`, `toString()` won't work

Reply:
good question. try testing with `ST_POINT(0, 0)` and see what's the literal BYTES constructed. 
but I think since we don't want to worry about these non-directly constructable literal types we can add a TODO and address later.

Reply:
Actually, the type can never be bytes. So I removed it.

for ST_POINT(0, 0), it will be passed in as string for literal.